pkg/cri/resource-manager/requests.go

```diff
@@ -189,6 +191,61 @@ func (m *resmgr) RunPod(ctx context.Context, method string, request interface{},
 	return reply, nil
 }
 
+// StopPod intercepts CRI requests for stopping Pods.
+func (m *resmgr) StopPod(ctx context.Context, method string, request interface{},
+	handler server.Handler) (interface{}, error) {
+
+	m.Lock()
+	defer m.Unlock()
+
+	podID := request.(*criapi.StopPodSandboxRequest).PodSandboxId
+	pod, ok := m.cache.LookupPod(podID)
+
+	if !ok {
+		m.Warn("%s: failed to look up pod %s, just passing request through", method, podID)
+	} else {
+		m.Info("%s: stopping pod %s (%s)...", method, pod.GetName(), podID)
+	}
+
+	reply, rqerr := handler(ctx, request)
+
+	if !ok {
+		return reply, rqerr
+	}
+
+	if rqerr != nil {
```

Comment:
should it handle errors like "already stopped" or similar conditions?

Reply:
If you mean if handling errors returned by the runtime like 'already stopped' for StopPodSandbox or StopContainer with a stopped pod's of container's ID, then there are no errors like that. The runtime will reply with success if you try to stop an already stopped container or pod.